internal/lib/lib.go

```diff
@@ -10,11 +10,11 @@ import (
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/policy"
 )
 
-// ResolveSubmitter will build out a ResultSubmitter if the provided pyxisClient, pc, is not nil.
-// The pyxisClient is a required component of the submitter. If pc is nil, then a noop submitter
-// is returned instead, which does nothing.
-func ResolveSubmitter(pc PyxisClient, projectID, dockerconfig, logfile string) ResultSubmitter {
-	if pc != nil {
+// ResolveSubmitter will build out a ResultSubmitter if the provided pyxisClient, pc, is not nil,
+// The pyxisClient is a required component of the submitter. If pc is nil or if konflux is false,
+// then a noop submitter is returned instead, which does nothing.
+func ResolveSubmitter(pc PyxisClient, projectID, dockerconfig, logfile string, konflux bool) ResultSubmitter {
```

Comment:
NIT: 

```suggestion
func ResolveSubmitter(pc PyxisClient, projectID, dockerconfig, logfile string, isKonflux bool) ResultSubmitter {
```

Reply:
Also, I think you can get by without changing this function signature.

When callers call ResolveSubmitter, they already have the context as to whether they're in a Konflux environment. As I understand it, they can just _skip calling ResolveSubmitter_ instead of passing that information into the check because they have the same information this function has.

Less code to change here, IMO.

Reply:
No, that's incorrect, this is needed for the case that a partner looks at our code and see's a env to bypass checks, and then passes in the flags to build a pyxis client. If this code was not there we would allow less checks to be executed by a partner that read our code. Basically this is a safeguard for that situation.

Reply:
I'm not sure I'm following. You can forcefully set a NoopSubmitter based on cfg.Konflux at the call point. I understand wanting to make sure that the partner doesn't find themselves in a situation they don't want, but the way I see it, it's the same precaution either way.

The way I see it, if we had followed a Konflux-only flow, we wouldn't be doing any submitter resolution at all, and would just be passing in a hard-coded NoopSubmitter.

Up to you; feel free to resolve if you think this is sufficient.